### ISSUE TITLE
flow crashes on OS X

### ISSUE THREAD

**Issue by aflatter** [NixOS/nixpkgs#16060]
## Issue description

The flow type checker does not run on OS X.

```
$ flow server
[2016-06-08 09:49:02] Initializing Server (This might take some time)
[2016-06-08 09:49:02] executable=/nix/store/vc9wczr2pvaj27420mj8snpigka3hr6j-flow-0.22.1/bin/flow
[2016-06-08 09:49:02] version=0.22.1
[2016-06-08 09:49:02] Parsing
[2016-06-08 09:49:02] Building package heap
[2016-06-08 09:49:02] Running local inference
[2016-06-08 09:49:02] Re-resolving directly dependent files
[2016-06-08 09:49:02] Calculating dependencies
[2016-06-08 09:49:02] Merging
[2016-06-08 09:49:02] Done
[2016-06-08 09:49:02] Server is READY
[2016-06-08 09:49:02] Took 0.631879 seconds to initialize.
Fatal error: exception End_of_file
```

I tried upgrading to a newer version. `0.24.2` successfully built, but showed the same behaviour. More recent versions would not build at all.
### Steps to reproduce
1. `nix-env -i flow`
2. `mkdir tmp && cd tmp`
3. `flow init`
4. `touch foo.js`
5. `flow server`
## Technical details
- System: OS X 10.11.5
- Nix version: 1.11.2
- Nixpkgs version: 16.09pre83147.df89584

**Comment by 2426021684** [comment]
For me flow does not even install
```
$ nix-env -i flow
installing ‘flow-0.30.0’
these derivations will be built:
  /nix/store/6343mn7zwh2q7w7jh194m7z75mzwdg6x-flow-0.30.0.drv
building path(s) ‘/nix/store/3rkli0wnpgar9azmcck4bs4zspvbszh5-flow-0.30.0’
unpacking sources
unpacking source archive /nix/store/9djka96b7hxwhbyvw7bdj0q78nv2b9xx-flow-v0.30.0-src
source root is flow-v0.30.0-src
patching sources
configuring
no configure script, doing nothing
building
make flags: SHELL=/nix/store/716mpwxqg1n49h34ayc8axx591sw88cm-bash-4.3-p42/bin/bash
mkdir -p bin
tar czf bin/flowlib.tar.gz -C lib .
rm -f _build/src/flow.d.byte _build/src/flow.native
rm -f _obuild/flow/flow.byte _obuild/flow/flow.asm
mkdir -p _build/hack/fsevents/
cp hack/fsevents/fsevents_stubs.c _build/hack/fsevents/fsevents_stubs.c
mkdir -p _build/hack/third-party/inotify/
cp hack/third-party/inotify/inotify_compat.h _build/hack/third-party/inotify/inotify_compat.h
mkdir -p _build/hack/third-party/libancillary/
cp hack/third-party/libancillary/ancillary.h _build/hack/third-party/libancillary/ancillary.h
cd _build/hack/fsevents/ && ocamlopt   -c fsevents_stubs.c
In file included from fsevents_stubs.c:37:
In file included from /nix/store/1svb87d4r83njgipcclb0cpaj85gazl9-apple-framework-CoreServices/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:39:
In file included from /nix/store/1svb87d4r83njgipcclb0cpaj85gazl9-apple-framework-CoreServices/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LaunchServices.h:19:
In file included from /nix/store/1svb87d4r83njgipcclb0cpaj85gazl9-apple-framework-CoreServices/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/IconsCore.h:23:
In file included from /nix/store/1svb87d4r83njgipcclb0cpaj85gazl9-apple-framework-CoreServices/Library/Frameworks/CoreServices.framework/Frameworks/OSServices.framework/Headers/OSServices.h:27:
In file included from /nix/store/1svb87d4r83njgipcclb0cpaj85gazl9-apple-framework-CoreServices/Library/Frameworks/CoreServices.framework/Frameworks/OSServices.framework/Headers/CSIdentity.h:43:
In file included from /nix/store/dcq5bmn754fv8c34l2vznfznh91xm0k3-apple-framework-Security/Library/Frameworks/Security.framework/Headers/Security.h:25:
/nix/store/dcq5bmn754fv8c34l2vznfznh91xm0k3-apple-framework-Security/Library/Frameworks/Security.framework/Headers/cssmconfig.h:51:18: error: typedef redefinition with different types ('uint64_t' (aka 'unsigned long long') vs 'unsigned long')
typedef uint64_t uint64;
                 ^
/nix/store/3ipfgq2y5q8zfid19lbbn60hh24xqj40-ocaml-4.01.0/lib/ocaml/caml/config.h:128:26: note: previous definition is here
typedef ARCH_UINT64_TYPE uint64;
                         ^
1 error generated.
make: *** [Makefile:170: _build/hack/fsevents/fsevents_stubs.o] Error 2
builder for ‘/nix/store/6343mn7zwh2q7w7jh194m7z75mzwdg6x-flow-0.30.0.drv’ failed with exit code 2
error: build of ‘/nix/store/6343mn7zwh2q7w7jh194m7z75mzwdg6x-flow-0.30.0.drv’ failed
```

**Comment by wereHamster** [comment]
@2426021684 your bug is fixed in ocaml 4.03, but that version is not yet in nixpkgs :(
